### ISSUE TITLE
%%%  does not work in JVM projects

### ISSUE THREAD

**Issue by Atry** [scala-js/scala-js#2979]
```
$ mkdir project
$ echo 'addSbtPlugin("org.scala-js" % "sbt-scalajs" % "0.6.16")' > project/plugins.sbt
$ echo 'libraryDependencies += "com.thoughtworks.extractor" %%% "extractor" % "2.1.0"' > build.sbt
$ sbt update
Updated file /private/tmp/project/build.properties setting sbt.version to: 0.13.15
[warn] Executing in batch mode.
[warn]   For better performance, hit [ENTER] to switch to interactive mode, or
[warn]   consider launching sbt without any commands, or explicitly passing 'shell'
[info] Loading global plugins from /Users/twer/.sbt/0.13/plugins
[info] Loading project definition from /private/tmp/project
[info] Updating {file:/private/tmp/project/}tmp-build...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
Reference to undefined setting: 

  *:isScalaJSProject from *:libraryDependencies (/private/tmp/build.sbt:1)
 
	at sbt.Init$class.Uninitialized(Settings.scala:270)
	at sbt.Def$.Uninitialized(Def.scala:10)
	at sbt.Init$class.delegate(Settings.scala:191)
	at sbt.Def$.delegate(Def.scala:10)
	at sbt.Init$class.compiled(Settings.scala:138)
	at sbt.Def$.compiled(Def.scala:10)
	at sbt.Init$class.make(Settings.scala:144)
	at sbt.Def$.make(Def.scala:10)
	at sbt.Load$$anonfun$8.apply(Load.scala:161)
	at sbt.Load$$anonfun$8.apply(Load.scala:156)
	at sbt.Load$.timed(Load.scala:1025)
	at sbt.Load$.apply(Load.scala:156)
	at sbt.Load$.defaultLoad(Load.scala:39)
	at sbt.BuiltinCommands$.liftedTree1$1(Main.scala:548)
	at sbt.BuiltinCommands$.doLoadProject(Main.scala:548)
	at sbt.BuiltinCommands$$anonfun$loadProjectImpl$2.apply(Main.scala:540)
	at sbt.BuiltinCommands$$anonfun$loadProjectImpl$2.apply(Main.scala:540)
	at sbt.Command$$anonfun$applyEffect$1$$anonfun$apply$2.apply(Command.scala:59)
	at sbt.Command$$anonfun$applyEffect$1$$anonfun$apply$2.apply(Command.scala:59)
	at sbt.Command$$anonfun$applyEffect$2$$anonfun$apply$3.apply(Command.scala:61)
	at sbt.Command$$anonfun$applyEffect$2$$anonfun$apply$3.apply(Command.scala:61)
	at sbt.Command$.process(Command.scala:93)
	at sbt.MainLoop$$anonfun$1$$anonfun$apply$1.apply(MainLoop.scala:96)
	at sbt.MainLoop$$anonfun$1$$anonfun$apply$1.apply(MainLoop.scala:96)
	at sbt.State$$anon$1.doX$1(State.scala:183)
	at sbt.State$$anon$1.process(State.scala:190)
	at sbt.MainLoop$$anonfun$1.apply(MainLoop.scala:96)
	at sbt.MainLoop$$anonfun$1.apply(MainLoop.scala:96)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.MainLoop$.next(MainLoop.scala:96)
	at sbt.MainLoop$.run(MainLoop.scala:89)
	at sbt.MainLoop$$anonfun$runWithNewLog$1.apply(MainLoop.scala:68)
	at sbt.MainLoop$$anonfun$runWithNewLog$1.apply(MainLoop.scala:63)
	at sbt.Using.apply(Using.scala:24)
	at sbt.MainLoop$.runWithNewLog(MainLoop.scala:63)
	at sbt.MainLoop$.runAndClearLast(MainLoop.scala:46)
	at sbt.MainLoop$.runLoggedLoop(MainLoop.scala:30)
	at sbt.MainLoop$.runLogged(MainLoop.scala:22)
	at sbt.StandardMain$.runManaged(Main.scala:109)
	at sbt.xMain.run(Main.scala:38)
	at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:109)
	at xsbt.boot.Launch$.withContextLoader(Launch.scala:128)
	at xsbt.boot.Launch$.run(Launch.scala:109)
	at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:35)
	at xsbt.boot.Launch$.launch(Launch.scala:117)
	at xsbt.boot.Launch$.apply(Launch.scala:18)
	at xsbt.boot.Boot$.runImpl(Boot.scala:41)
	at xsbt.boot.Boot$.main(Boot.scala:17)
	at xsbt.boot.Boot.main(Boot.scala)
[error] Reference to undefined setting: 
[error] 
[error]   *:isScalaJSProject from *:libraryDependencies (/private/tmp/build.sbt:1)
[error]  
[error] Use 'last' for the full log.
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? 
```

**Comment by sjrd** [comment]
Ah, I see. This happens if there is not a single Scala.js project in the build. If I add this line to `build.sbt`:
```scala
lazy val foo = project.enablePlugins(ScalaJSPlugin)
```
then it works. The `%%%` does work in the JVM-only project.

That's because the setup of `isScalaJSProject := false` is in [`ScalaJSPlugin.globalSettings`](https://github.com/scala-js/scala-js/blob/9393d7f47be5b0c9294e5c88c67271d98a125981/sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPlugin.scala#L396). It applies `in Global`, so works for JVM projects, but only if at least one project in the build enables `ScalaJSPlugin`.

I think we had that same problem in https://github.com/scala-native/sbt-crossproject at some point, but we fixed it in a different way than what is proposed in #2978. We added a separate `AutoPlugin` that is always enabled and that defines `crossPlatform` (equiv. `isScalaJSProject`) in `Global`. I think this solution is cleaner than using `.?`. I'll PR that shortly.
